src/NServiceBus.Core/Recoverability/SecondLevelRetries/SecondLevelRetries.cs

```diff
@@ -30,8 +33,10 @@ internal SecondLevelRetries()
         protected internal override void Setup(FeatureConfigurationContext context)
         {
             var  retryPolicy = GetRetryPolicy(context.Settings);
+            var slrStorage = new SlrStatusStorage();
 
             context.Container.RegisterSingleton(typeof(SecondLevelRetryPolicy), retryPolicy);
+            context.Container.RegisterSingleton(typeof(SlrStatusStorage), slrStorage);
```

Comment:
I think the reason that we register the storage in DI is because the satellites are reusing the FLR behavior (yes we should stop doing that). 

But I guess making that even more obvious is not a bad thing